sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/concatenate_data/concatenate_and_time_sync_node.hpp

```diff
@@ -82,115 +85,87 @@
-  std::vector<std::string> input_topics_;
-
-  /** \brief TF listener object. */
-  std::shared_ptr<tf2_ros::Buffer> tf2_buffer_;
-  std::shared_ptr<tf2_ros::TransformListener> tf2_listener_;
-
-  std::deque<geometry_msgs::msg::TwistStamped::ConstSharedPtr> twist_ptr_queue_;
-
-  std::map<std::string, sensor_msgs::msg::PointCloud2::ConstSharedPtr> cloud_stdmap_;
-  std::map<std::string, sensor_msgs::msg::PointCloud2::ConstSharedPtr> cloud_stdmap_tmp_;
+  struct Parameters
+  {
+    int maximum_queue_size;
+    double timeout_sec;
+    bool is_motion_compensated;
+    bool publish_synchronized_pointcloud;
+    bool keep_input_frame_in_synchronized_pointcloud;
+    bool publish_previous_but_late_pointcloud;
+    std::string synchronized_pointcloud_postfix;
+    std::string input_twist_topic_type;
+    std::vector<std::string> input_topics;
+    std::string output_frame;
+    std::vector<double> lidar_timestamp_offsets;
+    std::vector<double> lidar_timestamp_noise_window;
+  } params_;
+
+  double current_concatenate_cloud_timestamp_{0.0};
+  double lastest_concatenate_cloud_timestamp_{0.0};
+  bool drop_previous_but_late_pointcloud_{false};
+  bool publish_pointcloud_{false};
+  double diagnostic_reference_timestamp_min_{0.0};
+  double diagnostic_reference_timestamp_max_{0.0};
+  std::unordered_map<std::string, double> diagnostic_topic_to_original_stamp_map_;
+
+  std::shared_ptr<CombineCloudHandler> combine_cloud_handler_;
+  std::shared_ptr<CloudCollector> cloud_collector_;
+  std::list<std::shared_ptr<CloudCollector>> cloud_collectors_;
   std::mutex mutex_;
+  std::unordered_map<std::string, double> topic_to_offset_map_;
+  std::unordered_map<std::string, double> topic_to_noise_window_map_;
+
+  // subscribers
+  std::vector<rclcpp::Subscription<sensor_msgs::msg::PointCloud2>::SharedPtr> pointcloud_subs;
+  rclcpp::Subscription<geometry_msgs::msg::TwistWithCovarianceStamped>::SharedPtr twist_sub_;
+  rclcpp::Subscription<nav_msgs::msg::Odometry>::SharedPtr odom_sub_;
+
+  // publishers
+  rclcpp::Publisher<sensor_msgs::msg::PointCloud2>::SharedPtr concatenate_cloud_publisher_;
+  std::unordered_map<std::string, rclcpp::Publisher<sensor_msgs::msg::PointCloud2>::SharedPtr>
+    topic_to_transformed_cloud_publisher_map_;
+  std::unique_ptr<autoware::universe_utils::DebugPublisher> debug_publisher_;
 
-  std::vector<double> input_offset_;
-  std::map<std::string, double> offset_map_;
-
-  void transformPointCloud(const PointCloud2::ConstSharedPtr & in, PointCloud2::SharedPtr & out);
-  Eigen::Matrix4f computeTransformToAdjustForOldTimestamp(
-    const rclcpp::Time & old_stamp, const rclcpp::Time & new_stamp);
-  std::map<std::string, sensor_msgs::msg::PointCloud2::SharedPtr> combineClouds(
-    sensor_msgs::msg::PointCloud2::SharedPtr & concat_cloud_ptr);
-  void publish();
+  std::unique_ptr<autoware::universe_utils::StopWatch<std::chrono::milliseconds>> stop_watch_ptr_;
+  diagnostic_updater::Updater updater_{this};
```

Comment:
Renaming to `diagnostic_updater_` would be clearer

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/8300/commits/490b15cec2efa8ee8155746cc14100efcca40034